code/game/objects/items/weapons/cigs_lighters.dm

```diff
@@ -574,3 +574,198 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 			user << "<span class='warning'>You need to dry this first!</span>"
 	else
 		..()
+
+///////////////
+//VAPE NATION//
+///////////////
+/obj/item/clothing/mask/vape
+	name = "e-cigarette"
+	desc = "A classy and highly sophisticated electronic cigarette, for classy and dignified gentlemen. A warning label reads \"Warning: do not fill with flamable materials!\""//<<< i'd vape to that.
+	icon = 'icons/obj/clothing/masks.dmi'
+	icon_state = null
+	item_state = null
+	var/chem_volume = 100
+	var/vapetime = 0 //this so it won't puff out clouds every tick
+	var/screw = 0 // kinky
+	var/super = 0 //for the fattest vapes dude.
+	var/emagged = 0 //LET THE GRIEF BEGIN
+
+/obj/item/clothing/mask/vape/suicide_act(mob/user)
+	var/mob/living/carbon/human/H = user
```

Comment:
woops sorry, yeah, drop that, but be sure to check if the mob can have internal organs before touching said organs